compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -184,97 +171,40 @@ public STNode resumeParsing(ParserRuleContext context, Object... args) {
-                return parseObjectMethodOrField((STNode) args[0], (STNode) args[1]);
-            case OBJECT_FIELD_RHS:
-                return parseObjectFieldRhs((STNode) args[0], (STNode) args[1], (STNode) args[2], (STNode) args[3],
-                        (STNode) args[4]);
-            case OBJECT_TYPE_QUALIFIER:
-                return parseObjectTypeQualifiers();
-            case OBJECT_KEYWORD:
-                return parseObjectKeyword();
             case TYPE_NAME:
                 return parseTypeName();
-            case IF_KEYWORD:
-                return parseIfKeyword();
-            case ELSE_KEYWORD:
-                return parseElseKeyword();
-            case ELSE_BODY:
-                return parseElseBody();
-            case WHILE_KEYWORD:
-                return parseWhileKeyword();
-            case PANIC_KEYWORD:
-                return parsePanicKeyword();
-            case IMPORT_DECL_RHS:
-                return parseImportDecl((STNode) args[0], (STNode) args[1]);
-            case IMPORT_PREFIX:
-                return parseImportPrefix();
-            case IMPORT_MODULE_NAME:
-            case IMPORT_ORG_OR_MODULE_NAME:
-            case VARIABLE_REF:// 2 resume-points : parseQualifiedIdentifier(context)
-            case SERVICE_NAME:
+            case VARIABLE_REF:
             case IMPLICIT_ANON_FUNC_PARAM:
             case MAPPING_FIELD_NAME:
             case RECEIVE_FIELD_NAME:
-            case MODULE_ENUM_NAME:
             case ENUM_MEMBER_NAME:
                 return parseIdentifier(context);
-            case IMPORT_KEYWORD:
-                return parseImportKeyword();
             case SLASH:
                 return parseSlashToken();
             case DOT:
                 return parseDotToken();
-            case IMPORT_VERSION_DECL:
-                return parseVersion();
-            case VERSION_KEYWORD:
-                return parseVersionKeyword();
             case VERSION_NUMBER:
                 return parseVersionNumber();
             case DECIMAL_INTEGER_LITERAL:
             case MAJOR_VERSION:
             case MINOR_VERSION:
             case PATCH_VERSION:
                 return parseDecimalIntLiteral(context);
-            case IMPORT_SUB_VERSION:
-                return parseSubVersion(context);
-            case IMPORT_PREFIX_DECL:
-                return parseImportPrefixDecl();
-            case AS_KEYWORD:
-                return parseAsKeyword();
-            case CONTINUE_KEYWORD:
-                return parseContinueKeyword();
-            case BREAK_KEYWORD:
-                return parseBreakKeyword();
-            case RETURN_KEYWORD:
-                return parseReturnKeyword();
             case MAPPING_FIELD:
             case FIRST_MAPPING_FIELD:
                 return parseMappingField((ParserRuleContext) args[0]);
-            case SPECIFIC_FIELD_RHS:// 2 resume-points : parseMappingFieldRhs(key)
+            case SPECIFIC_FIELD_RHS:
```

Comment:
These mapping field related stuff are part of mapping constructor.Think these should go under resume expressions method ?

There are few other such cases.

Reply:
I ll look into it

Reply:
@m36dot There are still some more places that we can merge. eg: statements/module-level nodes related stuff that are added under "resumeOtherNodesParsing".
Is it possible to completely remove "resumeOtherNodesParsing" method?

Reply:
I am not sure if we can entirely remove resumeothernodesparsing method. Where should types and binding patterns be moved to?

Reply:
Can introduce a resume method for types?